### DISCUSSION TITLE
Coroutines as an alternative to actions?

### DISCUSSION THREAD

**Discussion by sojuz151** [libktx/ktx#397] [Ideas]
In korge engine you can use corutines to animate your object with suspend functions such as moveTo, you can read more about this here:
[https://docs.korge.org/korge/reference/animation/](https://docs.korge.org/korge/reference/animation/). 
In libGDX you can use action or manual keep state of your objects for such purpose but it would be nice to use corutines. 
What would be nice to have:
Suspend function runAction that will wait till action ( This can be easly implemented suspendCancellableCoroutine and RunAction)
Some suspend function such as moveTo
Maybe something fancy like tweens from korge.

**Comment by czyzby** [comment]
Yup, something like that should be possible to implement. Each action could be made available as a suspending function, so you could easily chain them in coroutines. Any code blocks would be executed via `Actions.run` (`RunnableAction`). I'd have to think about the parallel actions API, but otherwise it should be pretty straightforward. Thanks for the suggestion, I might look into it when I have more free time.
